# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLBlitHelper.h
# Commit: 4cff94ca9818
# Full Hash: 4cff94ca98189769f21a855520a345c6c4e5c841
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-15 03:50:53
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 044f8b594c71 -r 4cff94ca9818 gfx/gl/GLBlitHelper.h
--- a/gfx/gl/GLBlitHelper.h	Thu Jan 14 01:20:12 2021 +0000
+++ b/gfx/gl/GLBlitHelper.h	Thu Jan 14 01:23:06 2021 +0000
@@ -39,6 +39,7 @@
 class PlanarYCbCrImage;
 class SurfaceTextureImage;
 class MacIOSurfaceImage;
+class SurfaceDescriptor;
 class SurfaceDescriptorD3D10;
 class SurfaceDescriptorDXGIYCbCr;
 }  // namespace layers
@@ -194,6 +195,8 @@
   bool BlitImageToFramebuffer(layers::Image* srcImage,
                               const gfx::IntSize& destSize,
                               OriginPos destOrigin);
+  bool BlitSdToFramebuffer(const layers::SurfaceDescriptor&,
+                           const gfx::IntSize& destSize, OriginPos destOrigin);
 
  private:
   bool BlitImage(layers::GPUVideoImage* srcImage, const gfx::IntSize& destSize,